core-tests/src/test/java/org/mule/transformer/simple/AddSessionVariableTransformerTestCase.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.transformer.simple;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Matchers.argThat;
+import static org.mockito.Matchers.isNull;
+import static org.mockito.Mockito.RETURNS_DEEP_STUBS;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.mule.transformer.types.MimeTypes.APPLICATION_XML;
+import org.mule.api.MuleContext;
+import org.mule.api.MuleEvent;
+import org.mule.api.MuleMessage;
+import org.mule.api.MuleSession;
+import org.mule.api.expression.ExpressionManager;
+import org.mule.api.lifecycle.InitialisationException;
+import org.mule.api.metadata.DataType;
+import org.mule.api.metadata.SimpleDataType;
+import org.mule.api.transformer.TransformerException;
+import org.mule.tck.junit4.AbstractMuleTestCase;
+import org.mule.tck.junit4.matcher.DataTypeMatcher;
+import org.mule.tck.size.SmallTest;
+import org.mule.transformer.types.DataTypeFactory;
+import org.mule.transformer.types.MimeTypes;
+import org.mule.transformer.types.TypedValue;
+
+import javax.activation.MimeTypeParseException;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mockito;
+import org.mockito.internal.verification.VerificationModeFactory;
+
+@SmallTest
+public class AddSessionVariableTransformerTestCase extends AbstractMuleTestCase
```

Comment:
looks like there is a too much duplication between AddFlowVariableTransformerTestCase , AddPropertyTransformerTestCase and AddSessionVariableTransformerTestCase classes